### PR TITLE
.NET Interactive: Enable_try_dotnet_to_batch_5b

### PR DESCRIPTION
**Pull request by v-mepa**
Enable try dotnet to batch 5b

https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.contains | Only   first example
-- | --
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.single | All   examples can be enabled except for the second one
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.singleordefault | Except   for the last sample
https://docs.microsoft.com/en-us/dotnet/api/system.object.gettype |  
https://docs.microsoft.com/en-us/dotnet/api/system.string.chars | Except   for the first sample
https://docs.microsoft.com/en-us/dotnet/api/system.string.compareordinal | Except   for the first sample
https://docs.microsoft.com/en-us/dotnet/api/system.string.equals | Except   for first and last samples
https://docs.microsoft.com/en-us/dotnet/api/system.string.insert | Except   for last samples
https://docs.microsoft.com/en-us/dotnet/api/system.string.startswith | Except   for last two samples
https://docs.microsoft.com/en-us/dotnet/api/system.string.trimstart | Except   for last two samples

### CODE REVIEW

File: xml/System/String.xml [hunk 1]
```diff
@@ -2918,9 +2918,9 @@ Examples of instantiating strings:
 ## Examples
  This following example demonstrates that <xref:System.String.CompareOrdinal%2A> and <xref:System.String.Compare%2A> use different sort orders.
 
- [!code-cpp[StringCompareOrdinal#1](~/samples/snippets/cpp/VS_Snippets_CLR/StringCompareOrdinal/CPP/stringcompareordinal.cpp#1)]
- [!code-csharp[StringCompareOrdinal#1](~/samples/snippets/csharp/VS_Snippets_CLR/StringCompareOrdinal/CS/stringcompareordinal.cs#1)]
- [!code-vb[StringCompareOrdinal#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/StringCompareOrdinal/VB/stringcompareordinal.vb#1)]
+ :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_CLR/StringCompareOrdinal/CPP/stringcompareordinal.cpp" id="Snippet1":::
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR/StringCompareOrdinal/CS/stringcompareordinal.cs" interactive="try-dotnet" id="Snippet1":::
```
Comment (mairaw) [inline xml/System/String.xml:2922]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR/StringCompareOrdinal/CS/stringcompareordinal.cs" id="Snippet1":::
```

File: xml/System/String.xml [hunk 2]
```diff
@@ -14941,8 +14941,8 @@ The `TrimStart(System.Char[])` method removes from the current string all leadin
 
 The following example demonstrates the basic functionality of the <xref:System.String.TrimStart%2A> method:
 
-[!code-csharp[System.String.TrimStart#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs#1)]
-[!code-vb[System.String.TrimStart#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.TrimStart/vb/sample.vb#1)]
+::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs" interactive="try-dotnet-method" id="Snippet1":::
```
Comment (mairaw) [inline xml/System/String.xml:14944]:
```suggestion
:::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs" interactive="try-dotnet-method" id="Snippet1":::
```

File: xml/System.Linq/Enumerable.xml [hunk 3]
```diff
@@ -9405,14 +9405,14 @@ Only unique elements are returned.
...
+ 
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" interactive="try-dotnet-method" id="Snippet81":::
+ :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb" id="Snippet81":::
   
  The following code example demonstrates that <xref:System.Linq.Enumerable.Single%60%601%28System.Collections.Generic.IEnumerable%7B%60%600%7D%2CSystem.Func%7B%60%600%2CSystem.Boolean%7D%29> throws an exception when the sequence does not contain exactly one element that satisfies the condition.  
-  
- [!code-csharp[System.Linq.Enumerable#82](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs#82)]
- [!code-vb[System.Linq.Enumerable#82](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb#82)]  
+ 
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" interactive="try-dotnet-method" id="Snippet82":::
```
[the remark this answers is not in the document]
Comment (v-mepa) [reply xml/System.Linq/Enumerable.xml:9414]: Okay so I must number the examples as per the code sequence. I was numbering them as they appear on the web page. Bummer, yes this one is the second example in the code. Updating it now.

File: xml/System/String.xml [hunk 4]
```diff
@@ -1403,13 +1403,13 @@ Examples of instantiating strings:
...
-
+ 
 ## Examples
  The following example demonstrates how you can use this indexer in a routine to validate a string.
 
- [!code-cpp[Uri_IsHexDigit#1](~/samples/snippets/cpp/VS_Snippets_CLR/Uri_IsHexDigit/CPP/uri_ishexdigit.cpp#1)]
- [!code-csharp[Uri_IsHexDigit#1](~/samples/snippets/csharp/VS_Snippets_CLR/Uri_IsHexDigit/CS/uri_ishexdigit.cs#1)]
- [!code-vb[Uri_IsHexDigit#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/Uri_IsHexDigit/VB/uri_ishexdigit.vb#1)]
+ :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_CLR/Uri_IsHexDigit/CPP/uri_ishexdigit.cpp" id="Snippet1":::
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR/Uri_IsHexDigit/CS/uri_ishexdigit.cs" interactive="try-dotnet-method" id="Snippet1":::
```
[the remark this answers is not in the document]
Comment (v-mepa) [reply xml/System/String.xml:1411]: https://docs.microsoft.com/en-us/dotnet/api/system.string.chars | Except   for the first sample
-- | --


This API has only 1 example in it, I am committing the changes as you mentioned. Let me know if I am missing anything.

[the remark this answers is not in the document]
Comment (mairaw) [reply xml/System/String.xml:1411]: So what I meant is that the sample in the Remarks is ok to be interactive but not the one in the examples section.

[the remark this answers is not in the document]
Comment (v-mepa) [reply xml/System/String.xml:1411]: okay, I get it now :), I will add another commit.

File: xml/System/String.xml [hunk 5]
```diff
@@ -14941,8 +14941,8 @@ The `TrimStart(System.Char[])` method removes from the current string all leadin
 
 The following example demonstrates the basic functionality of the <xref:System.String.TrimStart%2A> method:
 
-[!code-csharp[System.String.TrimStart#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs#1)]
-[!code-vb[System.String.TrimStart#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.TrimStart/vb/sample.vb#1)]
+:::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs" interactive="try-dotnet-method" id="Snippet1":::
```
Comment (mairaw) [inline xml/System/String.xml:14944]: This code was supposed to work but we actually found a bug with the .NET interactive feature

```suggestion
:::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs" id="Snippet1":::
```

File: xml/System.Linq/Enumerable.xml [hunk 6]
```diff
@@ -9405,14 +9405,16 @@ Only unique elements are returned.
   
 ## Examples  
  The following code example demonstrates how to use <xref:System.Linq.Enumerable.Single%60%601%28System.Collections.Generic.IEnumerable%7B%60%600%7D%2CSystem.Func%7B%60%600%2CSystem.Boolean%7D%29> to select the only element of an array that satisfies a condition.  
-  
+ 
  [!code-csharp[System.Linq.Enumerable#81](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs#81)]
```
Comment (mairaw) [inline xml/System.Linq/Enumerable.xml:9409]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" interactive="try-dotnet-method" id="Snippet81":::
```

File: xml/System.Linq/Enumerable.xml [hunk 7]
```diff
@@ -9405,14 +9405,16 @@ Only unique elements are returned.
...
- [!code-vb[System.Linq.Enumerable#81](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb#81)]  
+ [!code-vb[System.Linq.Enumerable#81](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb#81)]
```
Comment (mairaw) [inline xml/System.Linq/Enumerable.xml:9410]:
```suggestion
 :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb" id="Snippet81":::
```

File: xml/System.Linq/Enumerable.xml [hunk 8]
```diff
@@ -9405,14 +9405,16 @@ Only unique elements are returned.
...
+
```
Comment (mairaw) [inline xml/System.Linq/Enumerable.xml:9411]:
```suggestion
```

**Pull request merged**